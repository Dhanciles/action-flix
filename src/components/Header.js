import React, { Component } from 'react'; 

export default const Header = () =>  {
    return (
      <header>
        <h1>Action Flix</h1>
        <button>Login</button>
      </header>
    )
}
