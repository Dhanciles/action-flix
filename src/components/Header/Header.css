.header {
  align-items: center;
  display: flex;
  justify-content: space-between;
  border-bottom: 6px solid #3D414C;
}

.header-title-navlink {
  text-decoration: none;  
}

.header-title {
  color: #76E1EF;
  font-size: 4rem;
  margin: 0;
  margin-left: 30px;
}

.login-btn {
  background-color: #76E1EF;
  border-radius: 15px;
  font-size: 1.3rem;
  height: 40px;
  margin-right: 30px;
  width: 120px
}

.welcome-user {
  color: #76E1EF;
  margin-right: 40px;
}