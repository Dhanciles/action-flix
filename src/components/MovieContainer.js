import React from 'react'
import { MovieCard } from './MovieCard'

export const MovieContainer = ({ movies }) => {
  const actionMovies = movies.map(movie => (<MovieCard {...movie} />))

  return (
    <div>
      {actionMovies}
    </div>
  )
}