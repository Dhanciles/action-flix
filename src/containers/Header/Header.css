.header {
  align-items: center;
  display: flex;
  justify-content: space-between;
  border-bottom: 6px solid #3D414C;
}

.header-title-navlink {
  text-decoration: none;  
}

.header-title {
  color: #76E1EF;
  font-size: 4rem;
  margin: 0;
  margin-left: 30px;
}

.login-btn {
  background-color: #76E1EF;
  border-radius: 15px;
  font-size: 1.3rem;
  height: 40px;
  margin-right: 30px;
  width: 120px
}

.dropdown-title {
  margin-right: 40px;
  position: relative;
}

.dropdown-title * {
  display: inline-block;
}

.dropdown-list {
  align-items: center;
  background-color: #fff;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 80px;
  margin-top: -15px;
  padding: 10px;
  position: absolute;
  width: 103px;
}

.favorites-btn,
.home-btn,
.logout-btn {
  border: none;
  font-family: 'Lato', sans-serif;
  font-size: 1rem;
  width: 100%;
  cursor: pointer;
}

.favorites-btn:hover,
.home-btn:dhover,
.logout-btn:hover {
  cursor: pointer;
  color: #76E1EF;
  text-shadow: 1px 1px #3D414C;
  cursor: pointer;
  text-decoration: underline;
} 

.arrow-icon {
  height: 10px;
  width: auto;
}

.welcome-user {
  cursor: pointer;
  color: #76E1EF;
  margin-right: 10px;
}