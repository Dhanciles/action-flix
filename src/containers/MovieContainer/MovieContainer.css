.movie-container {
  background-color: #323740;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  margin-top: 25px;
}