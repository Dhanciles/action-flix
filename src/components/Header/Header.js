import React, { Component } from 'react'; 
import { connect } from 'react-redux'
import { logOut } from '../../actions/index.js'
import { NavLink } from 'react-router-dom'
import './Header.css'
import PropTypes from 'prop-types'

export class Header extends Component {
  constructor() {
    super()
    this.state = {
      listOpen: false
    }
  }

  toggleList = () => {
    this.setState({
      listOpen: !this.state.listOpen
    })
  }

  handleLogOut = () => {
    this.toggleList()
    this.props.logUserOut()
  }

  render() {
    const arrowIcon = this.state.listOpen ? '../images/up-arrow.svg' : '../images/down-arrow.svg'

    const notLoggedIn = (
      <header className='header'>
        <NavLink to='/' className='header-title-navlink' ><h1 className='header-title'>Action Flix</h1></NavLink>
        <NavLink to='/login'><button className='login-btn'>Login</button></NavLink>
      </header>   
    )
  
    const loggedIn = (
      <header className="header">
        <h1 className='header-title'>Action Flix</h1>
          <div className='dropdown-wrapper'>
            <div className='dropdown-header'>
              <div className='dropdown-title' onClick={() => this.toggleList()}>
                <h3 className='welcome-user'>Hello {this.props.name}</h3>
                <img className='arrow-icon' src={arrowIcon} alt='down arrow' />
              </div>
            </div>
            { !this.state.listOpen ?  null : 
              <article className='dropdown-list'>
                <NavLink to='/'><button className='home-btn'>Home</button></NavLink>
                <NavLink to='/favorites'><button className='favorites-btn'>Favorites</button></NavLink>
                <button className='logout-btn' onClick={() => this.handleLogOut()}>Sign Out</button>
              </article>
              
            }
          </div>
      </header>
    )
    if (this.props.user) {
       return loggedIn
    }
    return (
        notLoggedIn 
    )
  }
}


Header.proptypes = {
  logOut: PropTypes.func.isRequired
}

export const mapStateToProps = (state) => ({
  user: state.userId,
  name: state.userName
})

export const mapDispatchToProps = (dispatch) => ({
  logUserOut: () => dispatch(logOut())
}) 

export default connect(mapStateToProps, mapDispatchToProps)(Header)

  