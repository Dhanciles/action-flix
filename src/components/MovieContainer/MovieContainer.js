import React, { Component } from 'react'
import { Route } from 'react-router-dom'
import MovieCard from '../MovieCard/MovieCard'
import './MovieContainer.css'
import { connect } from 'react-redux'
import { fetchMovies } from '../../thunks/fetchMovies.js'
import { movieDataBaseKey }  from '../../../src/constants.js'
import { getFavorites } from '../../thunks/getFavorites.js'


export class MovieContainer extends Component {
async componentDidMount() {
  let apiKey = movieDataBaseKey
  const url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&with_genres=28`
   await this.props.fetchMovies(url, this.props.userId)
}

render() {
  if (this.props.movies && this.props.location === '/') {
    const actionMovies = this.props.movies.map(movie => (<MovieCard {...movie} />))
    return (
      <div className='movie-container'>
        {actionMovies}
      </div>
    )
  } 
  // else if (this.props.movies && this.props.location === '/favorites') {
  //     const favoriteMovies = favoriteMoviesArr.map((movie) => (<MovieCard {...movie.props} />))

  //     return (
  //       <div className='movie-container'>
  //         {favoriteMovies}
  //       </div>
  //     )
  //   } 
    else {
      return <div>{'error'}</div>
    }
  }
}

export const mapStateToProps = (state) => {
  return {
  userId: state.userId,
  movies: state.movies,
  isLoading: state.isLoading,
  hasErrored: state.hasErrored
}}

export const mapDispatchToProps = (dispatch) => ({
  fetchMovies: (url, userId) => dispatch(fetchMovies(url, userId)),
  getFavorites: (userId) => dispatch(getFavorites(userId)) 
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieContainer)
