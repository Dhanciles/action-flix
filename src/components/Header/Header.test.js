import React from 'react'
import { shallow, mount } from 'enzyme'
import { Header } from './Header'
import { mapDispatchToProps } from '../MovieContainer/MovieContainer';

describe('Header', () => {
  let wrapper
  let mockLogOut
  beforeEach(() => {
    mockLogOut = jest.fn()
    wrapper = shallow(<Header logOutUser={ mockLogOut }/>)
  })
  it('should match the snapshot', () => {
    
    expect(wrapper).toMatchSnapshot();
  });
  describe('mapDispatchToProps', () => {
    it('should call dispatch when action is dipatched', () => {
      const mockDispatch = jest.fn()
      const isLoggedOut = mapDispatchToProps(mockDispatch)
      
      expect(isLoggedOut).toHaveBeencalled

    })
  })

  it.skip('should handle a click event that will log out our user', () => {

  })
})