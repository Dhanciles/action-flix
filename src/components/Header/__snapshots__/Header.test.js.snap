// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logOutUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header-title"
        >
          Action Flix
        </h1>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/login"
        >
          <button
            className="login-btn"
          >
            Login
          </button>
        </NavLink>,
        <button
          className="login-btn"
          onClick={[Function]}
        >
          Sign Out
        </button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Action Flix",
          "className": "header-title",
        },
        "ref": null,
        "rendered": "Action Flix",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            className="login-btn"
          >
            Login
          </button>,
          "to": "/login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login-btn",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Out",
          "className": "login-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header-title"
          >
            Action Flix
          </h1>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/login"
          >
            <button
              className="login-btn"
            >
              Login
            </button>
          </NavLink>,
          <button
            className="login-btn"
            onClick={[Function]}
          >
            Sign Out
          </button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Action Flix",
            "className": "header-title",
          },
          "ref": null,
          "rendered": "Action Flix",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="login-btn"
            >
              Login
            </button>,
            "to": "/login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "login-btn",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Out",
            "className": "login-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
